src/drivers/actuators/vertiq_io/vertiq_io.cpp

```diff
@@ -0,0 +1,306 @@
+		}
+	}
+
+	//Get the most up to date version of our parameters/check if they've changed
+	parameters_update();
+
+	//Make sure we also update the mixing output to get the most up to date configuration
+	_mixing_output.update();
+	_mixing_output.updateSubscriptions(true);
+
+	//Go ahead and check to see if our actuator test has gotten anything new
+	if (_actuator_test_sub.updated()) {
+		_actuator_test_sub.copy(&_actuator_test);
+
+		//Our test is active if anyone is giving us commands through the actuator test
+		_actuator_test_active = _actuator_test.action == actuator_test_s::ACTION_DO_CONTROL;
+	}
+
+	//stop our timer
+	perf_end(_loop_perf);
+}
+
+
+void VertiqIo::parameters_update()
+{
+	//If someone has changed any parameter in our module. Checked at 1Hz
+	if (_parameter_update_sub.updated()) {
+		//Grab the changed parameter with copy (which lowers the "changed" flag)
+		parameter_update_s param_update;
+		_parameter_update_sub.copy(&param_update);
+
+		// If any parameter updated, call updateParams() to check if
+		// this class attributes need updating (and do so).
+		updateParams();
+
+		//If the target module ID changed, we need to update all of our parameters
+		if (_param_vertiq_target_module_id.get() != _client_manager.GetObjectIdNow()) {
+			_client_manager.UpdateClientsToNewObjId(_param_vertiq_target_module_id.get());
+
+			//Make sure we start a new read!
+			_param_vertiq_trigger_read.set(true);
+			_param_vertiq_trigger_read.commit_no_notification();
+		}
+
+		//If you're set to re-read from the motor, mark all of the IQUART parameters for reinitialization, reset the trigger, and then update the IFCI params
+		if (_param_vertiq_trigger_read.get()) {
+			_client_manager.MarkIquartConfigsForRefresh();
+			_param_vertiq_trigger_read.set(false);
+			_param_vertiq_trigger_read.commit_no_notification();
+		}
+
+		_client_manager.UpdateIquartConfigParams();
+	}
+}
+
+bool VertiqIo::updateOutputs(bool stop_motors, uint16_t outputs[MAX_ACTUATORS], unsigned num_outputs,
+			     unsigned num_control_groups_updated)
+{
+#ifdef CONFIG_USE_IFCI_CONFIGURATION
+
+	if (_mixing_output.armed().armed) {
+
+		if (_param_vertiq_arm_behavior.get() == FORCE_ARMING && _send_forced_arm) {
+			_client_manager.SendSetForceArm();
+			_send_forced_arm = false;
+		}
+
+		//We already get a mixer value from [0, 65535]. We can send that right to the motor, and let the input parser handle
+		//conversions
+		_motor_interface_ptr->BroadcastPackedControlMessage(*_serial_interface.get_iquart_interface(), outputs, _cvs_in_use,
+				_telemetry_request_id);
+
+		//We want to make sure that we send a valid telem request only once to ensure that we're not getting extraneous responses.
+		//So, here we'll set the telem request ID to something that no one will respond to. Another function will take charge of setting it to a
+		//proper value when necessary
+		_telemetry_request_id = _impossible_module_id;
+
+	} else if (_actuator_test_active) {
```

Comment:
I don't care about forcing arming if we're using the actuator test. just spit out whatever values it's giving us